### ISSUE TITLE
PreviousMessageListQuery.load always return the same set of result

### ISSUE THREAD

**Issue by stevengpwc** [sendbird/SendBird-SDK-JavaScript#128]
I am writing a React app. I have separated the calls of createPreviousMessageListQuery and load. 

```javascript
  static createPreviousMessageListQuery(channel: GroupChannel): PreviousMessageListQuery {
    const queryInstance = channel.createPreviousMessageListQuery();
    queryInstance.limit = 30;
    queryInstance.reverse = false; // earlist comes at first and the latest at last
    return queryInstance;
  }

  static async getPreviousMessages(
    queryInstance: PreviousMessageListQuery,
  ): Promise<UserMessage[]> {
    return new Promise((resolve, reject) => {
      queryInstance.load((messages, error) => {
        if (error) {
          console.error(error);
          resolve([])
        }
        resolve(messages as UserMessage[]);
      })
    })
  }
```

I save the query instance in Context. Whenever I need to load more previous messages, I get back the instance from Context, and then call load() with that instance. But it always return the same set of messages.

The `sendbird` package version I am using is 3.0.111

**Comment by stevengpwc** [comment]
I switched to use the deprecated `createMessageListQuery(): MessageListQuery` which works for me. But the interesting thing is, the values in `messageList` and `error` in the callback of prev() call are swapped.

**Comment by aibek-bird** [comment]
Hi @stevengpwc , could you please answer me following question. Did you use the `setErrorFirstCallback` method? You can do that with `SendBird.getInstance().setErrorFirstCallback(true)` or you can specify `false` value as well. If it is true, then the error object will always come first. Please let me know if this solves your issue

**Comment by stevengpwc** [comment]
From 11/18/2019 to 11/22/2019

Steve KP Ng is out of office from 11/18/2019 to 11/22/2019 inclusive.

--Comments--


----------------------------------------------------------------------
The information transmitted is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. Any review, retransmission, dissemination or other use of, or taking of any action in reliance upon, this information by persons or entities other than the intended recipient is prohibited. If you received this in error, please contact the sender and delete the email together with any material attached (if any) completely from any device immediately. Any views or opinions expressed in this email are solely those of the author and do not necessarily represent those of PwC. 

Your privacy is of utmost importance to us. For more information on how we collect and process your personal information, please see our Privacy Statement at https://www.pwchk.com/en/privacy.html.

If you would like to subscribe to our e-newsletter, update your contact details or customise the information you receive from us, please complete the form at https://www.pwchk.com/en/subscribe-to-e-newsletter.html. If you wish to be removed from our distribution list, please click https://www.pwchk.com/en/unsubscribe.html to unsubscribe.

**Comment by anthonyemery** [comment]
This has just started happening to me too, where previously it worked? Do we have a regression here?

**Comment by aibek-bird** [comment]
Hi @anthonyemery, can you please provide me more details such as code snippet, if you use `setErrorFirstCallback` method and output? Looking into our codebase, there seems to be no problem.

**Comment by anthonyemery** [comment]
Hi, I do not use `setErrorFirstCallback`

I have discovered that if I downgrade to `3.0.108` then it works as expected once again

**Comment by stevengpwc** [comment]
Same here, I do not use setErrorFirstCallback

**Comment by aibek-bird** [comment]
Hi, @anthonyemery , @stevengpwc can you please try `setErrorFirstCallback(true)` method, and let us know if it works. The error comes last in the callbacks by default for our SDK to allow backward compatibility, and if you want to come it first, you can use the method above. Also, FYI you can use Promise patterns with newer versions of the SDK

**Comment by cooperjbrandon** [comment]
I see this error as well I'm using version `3.0.112`.

It looks like the timestamp on the request is always `9007199254740991`.

It looks like this is because (from the minified code), after a response from the sendbird, the code is:
```
                  ze.set(o, Math.min(t.reduce(function (e, n) {
                    return e > n.createdAt ? e : n.createdAt;
                  }, Number.MAX_SAFE_INTEGER), r));
```

where `ze` looks like a `Map` object , and `o` is the `PreviousMessageListQuery`. `t` in this case are the messages returned from sendbird. It seems like sendbird loops through each message, and if the `createdAt` timestamp is greater than the accumulator, sendbird uses that message's timestamp as the new accumulator value. However, `Number.MAX_SAFE_INTEGER` is `9007199254740991`, so that's always going to be greater than `n.createdAt`, so the value returned from `t.reduce` is `9007199254740991`.

Also, higher up in the code, sendbird does: `ze.set(this, Number.MAX_SAFE_INTEGER)`. So in this case, `r` is also `Number.MAX_SAFE_INTEGER`. Therefore, `Math.min(` returns `9007199254740991` every time.

And that's the value that's used in the query-param when loading more messages. I.e: `message_ts=9007199254740991`.

**Comment by cooperjbrandon** [comment]
Also note that the issue disappears for me after downgrading to `3.0.103`.

**Comment by aibek-bird** [comment]
@anthonyemery @cooperjbrandon @stevengpwc This bug was fixed in v3.0.113 (latest) Please let me know if this was solved for you.

**Comment by shagun121** [comment]
i am still facing the same issue on the latest version please help!!
